File: 23_boj_1837.cpp
#include <cstdio>
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>
#define MAX 1000000

using namespace std;

int prime[MAX+1]={0,}; // 0 => prime, 1 => composite

void eratosthenes() {
    for (int i=2; i*i<MAX; i++) {
        if (prime[i]==0) {
            for (int j=i*i; j<MAX; j+=i) {
                prime[j]=1;
            }
        }
    }
}

char p[100];
int k;

int main() {
    eratosthenes();
    scanf("%s %d", p, &k);

    for (int i=2; i<k; i++) {
        if (prime[i]==1) {
            continue;
        }
        int tmp=0;
        for (int j=0; p[j]; j++) {
            tmp=(tmp*10+(p[j]-'0'))%i;
        }
        if ( tmp==0 ) {
            printf("BAD %d\n",i);
            return 0;
        }
    }
    printf("GOOD\n");
    return 0;
}

File: 23_boj_1837.py
MAX_INT = 1000000

prime = [0 for i in range(MAX_INT+1)] # 0=>prime / 1=>composite

def eratosthenes() :
    for i in range(2,1000) :
        if prime[i]==0 :
            j=i*i
            while (True) :
                if j>MAX_INT :
                    break
                prime[j]=1
                j+=i

eratosthenes()

p,k=map(int,input().split())

for i in range(2,MAX_INT+1) :
    if (i >= k) :
        print("GOOD")
        exit()
    if prime[i] == 0 :
        if p % i == 0 :
            print("BAD "+str(i))
            exit()